crates/edr_provider/src/data.rs

```diff
@@ -948,6 +944,7 @@ impl<LoggerErrorT: Debug> ProviderData<LoggerErrorT> {
 
         let snapshot = Snapshot {
             block_number: self.blockchain.last_block_number(),
+            block_state_cache: self.block_state_cache.clone(),
```

Comment:
Add the cache to the snapshot to ensure that people can revert to previous snapshot (excluding newer cache entries).